Stratis.Bitcoin/Features/Consensus/ConsensusFeature.cs

```diff
@@ -180,7 +180,9 @@ private void RunLoop()
 
                 // TODO Need to revisit unhandled exceptions in a way that any process can signal an exception has been
                 // thrown so that the node and all the disposables can stop gracefully.
+                this.logger.LogDebug("Exception occurred in consensus loop: {0}", ex.ToString());
                 this.logger.LogCritical(new EventId(0), ex, "Consensus loop unhandled exception (Tip:" + this.consensusLoop.Tip?.Height + ")");
+                NLog.LogManager.Flush();
```

Comment:
This is not the right place to do that.

Reply:
this leads to app crash, if i don't do it here i don't do it ever
this is not elegant, but since we are using exceptions, there is little we can do about it 

this is the highest level in the thread running this code, so there is nowhere else this could be done

Reply:
hmm I didn't understand can you not just do it in the ConsensusFeatire.Stop()?

Reply:
no because exception here crashes the node, there is no normal shutdown, this is hard crash